subprojects/core/src/integTest/groovy/org/gradle/api/tasks/TaskDefinitionIntegrationTest.groovy

```diff
@@ -633,4 +633,75 @@ class TaskDefinitionIntegrationTest extends AbstractIntegrationSpec {
         then:
         failure.assertHasCause("Adding a task provider directly to the task container is not supported.")
     }
+
+    def "can override task name, description and group without @Internal annotation"() {
+        buildFile << """
+            class CustomTask extends DefaultTask {
+                @Override
+                public String getName() {
+                    return "customTask";
+                }
+                @Override
+                public String getDescription() {
+                    return "My custom task description.";
+                }
+                @Override
+                public String getGroup() {
+                    return "custom group";
+                }
+                @TaskAction
+                void doWork() {
+                    println("Hello from CustomTask")
+                }
+            }
+            tasks.register("customTask", CustomTask)
+        """
+
+        when:
+        succeeds("tasks")
+
+        then:
+        outputContains("""
+Custom group tasks
+------------------
+customTask - My custom task description.
+""")
+
+        when:
+        succeeds("customTask")
+
+        then:
+        outputContains("Hello from CustomTask")
+    }
+
+    // This test just shows the problem with overriding the task name,
+    // if it does not match name at the registration it will not be found.
+    def "fails when running a task with overriden name that is registered with a different name"() {
```

Comment:
I don't think we should allow overriding the task name. We should instead make that method `final` in 8.0, and just ignore the problem until them.

Reply:
Ok, I can remove this and I will open an issue to make it final in 8.0.

Reply:
Removed this test.